src/main/java/com/example/gamelist/GameService.java

```diff
@@ -0,0 +1,26 @@
+package com.example.gamelist;
+
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class GameService {
+
+    private final GameMapper gameMapper;
+
+    public GameService(GameMapper gameMapper) {
+        this.gameMapper = gameMapper;
+    }
+
+    public void updateGame(final Integer id, String name) {
+        Game game = gameMapper.findById(id).orElseThrow(() -> new GameNotFoundException("game information not found"));
+        List<Game> otherGames = this.gameMapper.findByName(name);
+
+        if (otherGames.stream().anyMatch(game -> game.getName().equals(name))) {
+            game.setName(UpdateRequest.getName());
```

Comment:
ServiceにはupdateGameメソッドの引数として name を渡していますね。
これをService内では更新後の名前としてつかいますのでUpdateRequestは不要です。
```suggestion
            game.setName(name);
```

Reply:
34b6bb543d4df47c0cc2b413f1b9d050a2df5085
<img width="739" alt="スクリーンショット 2023-12-29 23 59 17" src="https://github.com/otyacazyu/gamelist/assets/143002834/c835841d-8945-4aeb-b217-02f5bce83e0f">

引数や戻り値などがきちんと覚えられてないようで、もう一度復習を行おうと思います。

Reply:
そうですね。
引数や戻り値大事。

こちらResolveしてよいです。